src/components/molecules/ChatList/ChatList.tsx

```diff
@@ -0,0 +1,171 @@
+import React, { useState } from "react";
+import { User } from "types/User";
+import UserProfileModal from "components/organisms/UserProfileModal";
+import {
+  PrivateChatMessage,
+  RestrictedChatMessage,
+} from "components/context/ChatContext";
+import { useSelector } from "hooks/useSelector";
+import { WithId } from "utils/id";
+import { Modal } from "react-bootstrap";
+import { DEFAULT_PARTY_NAME, DEFAULT_PROFILE_IMAGE } from "settings";
+import { formatUtcSeconds } from "utils/time";
+import { getLinkFromText } from "utils/getLinkFromText";
+import { useUser } from "hooks/useUser";
+import "./ChatList.scss";
+
+interface ChatListProps {
+  messages: WithId<RestrictedChatMessage | PrivateChatMessage>[];
+  emptyListMessage?: string;
+  allowDelete?: boolean;
+  deleteMessage: (id: string) => Promise<void>;
+}
+
+const ChatList: React.FC<ChatListProps> = ({
+  messages,
+  allowDelete,
+  emptyListMessage,
+  deleteMessage,
+}) => {
+  const { user } = useUser();
+  const usersById = useSelector((state) => state.firestore.data.chatUsers);
```

Comment:
Hm, `useSelector` shouldn't be so problematic in its default usage case, surely? How certain are we that this will cause re-renders?

I'm fairly sure the "return object" pattern causes re-renders, but not sure about the above. This one has potential shallow-compare issues:

```
const { a, b } = useSelector((state) => ({a: state.a, b: state.b}));
```

Reply:
It does cause re-renders, that's why reselect was being used before the hooks. For the memo.

Reply:
> Hm, `useSelector` shouldn't be so problematic in its default usage case, surely? How certain are we that this will cause re-renders?

I haven't deeply dissected it, but as per the documentation links in https://github.com/sparkletown/sparkle/pull/718#discussion_r524727812 it (if not somewhat cryptically) seems to suggest the things I have called out here as potential culprits for breaking memoisation, if not re-rendering.

Reply:
@denisdimitrov this one probably still needs to be extracted as a selector